webapp/client/src/components/FormFieldEuroInput.js

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { IMaskInput } from "react-imask";
+import classNames from "classnames";
+import FormFieldScaffolding from "./FormFieldScaffolding";
+import FieldLabel from "./FieldLabel";
+import FormFieldSeparatedField from "./FormFieldSeparatedField";
+
+function FormFieldEuroInput({
+  fieldName,
+  fieldId,
+  value,
+  required,
+  autofocus,
+  label,
+  currency,
+  fieldWidth,
+  maxLength,
+  details,
+  errors,
+}) {
+  const calculatedMaxLength =
+    maxLength !== undefined ? maxLength + Math.floor(maxLength / 3) : undefined;
```

Comment:
@VictorDelCampo @SannyNguyenHung What do you think about this? Does this make sense. The case that we do not prevent from happening is a max length of 5 to allow `1234,5` as an input. Because this would result in 1.234, and the ',' and '.' are both taken into account.
I wouldn't want to add +1 for the comma because this can result in a different behavior in the other direction:
- max-length: 5
- input: 123456
- output: 123.456

What do you think?

Reply:
If I have a maxLength of '7' the result will be '9'. I really don't get it:

`maxLength + Math.floor(maxLength / 3)`
=
`7 + 7/3`

How will that calculate the correct max length? Sorry :see_no_evil:

Reply:
We had some misscommunication, I talked to Julian about dynamically expanding the maxLength of the input field, if:

- the user entered a "," for the fractional part of the number
- if a thousand-separator was automatically added

Both are taken into account for the maxLength if the number is input by the user number by number.

If the number is pasted, only the "," is taken into account and not the thousand-operator

Reply:
Why do you need a math.floor for the maxLength? maybe you want to use it with the given value? 
The problem what I can see is, that we should not calculate the maxLength based on the given value (side effects or behavior as you describe Julian).

Maybe we should leave it as it is...